imports/wasi_snapshot_preview1/fs.go

```diff
@@ -710,18 +710,16 @@ func fdReaddirFn(ctx context.Context, mod api.Module, params []uint64) Errno {
 	// Now, write entries to the underlying buffer.
 	if bufused > 0 {
 
-		// Cookie is zero when there is no cookie. Now that we know we will
-		// write entries, ensure the first one is one not zero.
-		if cookie == uint64(0) {
-			cookie = 1
-		}
+		// d_next is the index of the next file in the list, so it should
+		// always be one higher than the requested cookie.
+		d_next := cookie + 1
```

Comment:
I renamed this to d_next so that it's more clear we're not talking about the initial cookie anymore, but about the d_next value of the entries that we're writing away. 

I changed this to always be +1 because d_next is the cookie of the next record, so if you ask cal fd_readdir with cookie 7, you should never get an entry that has a d_next value of 7.

Reply:
great comment